qiskit/aqua/operators/primitive_ops/primitive_op.py

```diff
@@ -264,6 +271,8 @@ def to_circuit_op(self) -> OperatorBase:
         """ Returns a ``CircuitOp`` equivalent to this Operator. """
         # pylint: disable=import-outside-toplevel
         from .circuit_op import CircuitOp
+        if self.coeff == 0:
+            return CircuitOp(QuantumCircuit(self.num_qubits), coeff=0)
```

Comment:
Why is this special case needed? Doesn't the following line capture this case as well?

Reply:
This is needed when the operator is not unitary (cannot be converted into QuantumCircuit).